hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ScmTableCountTask.java

```diff
@@ -0,0 +1,212 @@
+  private final DSLContext dslContext;
+  private final long interval;
+  private ReadWriteLock lock = new ReentrantReadWriteLock(true);
+  private static final long NANOSECONDS_IN_MILLISECOND = 1_000_000;
+
+  public ScmTableCountTask(ReconStorageContainerManagerFacade reconSCM,
+                           ReconTaskStatusDao reconTaskStatusDao,
+                           ReconTaskConfig reconTaskConfig,
+                           ScmTableCountDao scmTableCountDao,
+                           UtilizationSchemaDefinition schemaDefinition) {
+    super(reconTaskStatusDao);
+    this.scmDBStore = reconSCM.getScmDBStore();
+    this.scmTableCountDao = scmTableCountDao;
+    this.dslContext = schemaDefinition.getDSLContext();
+    this.interval = reconTaskConfig.getScmTableCountTaskInterval().toMillis();
+  }
+
+  @Override
+  protected synchronized void run() {
+    try {
+      while (canRun()) {
+        wait(interval);
+        runTableCountProcess();
+      }
+    } catch (Throwable t) {
+      LOG.error("Error while running ScmTableCountTask: {}", t);
+      if (t instanceof InterruptedException) {
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  private void runTableCountProcess() {
+      try {
+        long startTime, endTime, duration, durationMilliseconds;
+
+        int execute = dslContext.truncate(SCM_TABLE_COUNT_TABLE_NAME).execute();
+        LOG.info("Deleted {} records from {}", execute,
+            SCM_TABLE_COUNT_TABLE_NAME);
+
+        startTime = System.nanoTime();
+        processTableCount();
+        endTime = System.nanoTime();
+        duration = endTime - startTime;
+        durationMilliseconds = duration / NANOSECONDS_IN_MILLISECOND;
+        LOG.info(
+            "Elapsed Time in milliseconds for processTableCount() execution: {}",
+            durationMilliseconds);
+      } catch (Exception e) {
+        LOG.error("An error occurred while performing table count: {}", e);
+      }
+    }
+
+  /**
+   * Processes the table count by iterating over SCM tables and retrieving the
+   * counts of objects in each table. The counts are then stored in the Recon
+   * database.
+   *
+   * @throws IOException if an I/O error occurs during table count processing.
+   */
+  public void processTableCount() throws IOException {
+    // Acquire write lock
+    lock.writeLock().lock();
+    try {
+      // Initialize the object count map
+      Map<String, Long> objectCountMap = initializeCountMap();
+
+      // Iterate over SCM tables
+      for (String tableName : getTaskTables()) {
+        Table table = scmDBStore.getTable(tableName);
+
+        try (TableIterator keyIter = table.iterator()) {
+          // Retrieve the count of objects in the table
+          long count = Iterators.size(keyIter);
+          objectCountMap.put(getRowKeyFromTable(tableName), count);
+        } catch (IOException ioEx) {
+          LOG.error("Unable to populate SCM Table Count in Recon DB.", ioEx);
```

Comment:
Can we mention the table name for which we got error ?

Reply:
Done